lib/action_client/base.rb

```diff
@@ -3,34 +3,55 @@ class Base < AbstractController::Base
     include AbstractController::Rendering
     include ActionView::Layouts
 
-    cattr_accessor :defaults,
+    class_attribute :defaults,
       instance_accessor: true,
       default: ActiveSupport::OrderedOptions.new
 
+    class_attribute :middleware,
+      instance_accessor: true
+
+    class_attribute :after_actions,
+      instance_accessor: true,
+      default: []
+
     class << self
+      def inherited(descendant)
+        descendant.middleware = Rails.configuration.action_client.middleware.dup
+        descendant.after_actions = after_actions.dup
+      end
+
       def default(options)
         options.each do |key, value|
           defaults[key] = value
         end
       end
 
+      def after_action(&block)
```

Comment:
This also borrows inspiration from ActionController's [`after_action`](https://api.rubyonrails.org/classes/AbstractController/Callbacks/ClassMethods.html#method-i-after_action), but _is_ in fact, different.

I considered aliasing this to a few alternatives, including `after_submit`.

What do you all think? Should we try to make our naming choices resemble existing Rails concepts, or is that setting us up for confusion?

Reply:
I think I like `after_submit` more. There's a risk of an uncanny valley effect if we almost follow the Rails interface, but not exactly. The most difference that might most easily trip someone up is the arity of the block; a different method name could help solidify the distinction.